### PR TITLE
fixes #2693 - don't cause handle_ca error when no Puppet CA associated with host

### PR DESCRIPTION
**Pull request by domcleal**

### CODE REVIEW

**Comment by GregSutcliffe** [comment]
Untested, but looks sane to me. Over to you Jenkins ;)

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -221,11 +221,9 @@ def pm_fqdn
   # Called after a host is given their provisioning template
   # Returns : Boolean status of the operation
   def handle_ca
-    return true if Rails.env == "test"
     return true unless Setting[:manage_puppetca]
-    if puppetca?
-      respond_to?(:initialize_puppetca,true) && initialize_puppetca && delCertificate && setAutosign
-    end
+    return true unless puppetca?
+    respond_to?(:initialize_puppetca,true) && initialize_puppetca && delCertificate && setAutosign
```
Comment (ohadlevy) [inline app/models/host/managed.rb:226]: what am I missing? that it returned false before?

Comment (domcleal) [inline app/models/host/managed.rb:226]: it returned nil before, which caused the controller to think it'd failed.  You wrote this patch :)

Comment (domcleal) [inline app/models/host/managed.rb:226]: see https://github.com/theforeman/foreman/blob/develop/app/controllers/unattended_controller.rb#L162 and http://projects.theforeman.org/issues/2693#note-1

Comment (ohadlevy) [inline app/models/host/managed.rb:226]: in that case :+1:

**Comment by domcleal** [comment]
@ohadlevy ignore the test failures, they're due to lack of disk space on slaves.

I updated the commit only to remove some trailing whitespace from the patch.

**Comment by domcleal** [comment]
ping @ohadlevy

**Comment by ohadlevy** [comment]
merged as f09d61d - thanks!

**Pull request closed without merging**